src/main/webapp/app/entities/submission.model.ts

```diff
@@ -76,6 +76,20 @@ export function getSubmissionResultByCorrectionRound(submission: Submission | un
     return undefined;
 }
 
+/**
+ * Used to access a submissions result for a specific id
+ *
+ * @param submission
+ * @param resultId
+ * @returns the results or undefined if submission or the result for the requested id is undefined
+ */
+export function getSubmissionResultById(submission: Submission | undefined, resultId: number): Result | undefined {
+    if (submission && submission.results) {
+        return submission.results.find((result) => result.id === resultId);
+    }
+    return undefined;
+}
```

Comment:
you can simplify it with this:

```suggestion
    return submission?.results?.find((result) => result.id === resultId);
}
```